src/render/draw_debug.js

```diff
@@ -210,3 +215,96 @@ function drawTextToOverlay(painter: Painter, text: string) {
     painter.debugOverlayTexture.update(canvas);
     painter.debugOverlayTexture.bind(gl.LINEAR, gl.CLAMP_TO_EDGE);
 }
+
+let debugCanvas: ?HTMLCanvasElement;
+let aabbCorners = [];
+
+function initializeCanvas(tr: Transform) {
+    if (!debugCanvas) {
+        debugCanvas = window.document.createElement('canvas');
+        window.document.body.appendChild(debugCanvas);
+        debugCanvas.style.position = 'absolute';
+        debugCanvas.style.left = 0;
+        debugCanvas.style.top = 0;
+        debugCanvas.style.pointerEvents = 'none';
+
+        const resize = () => {
+            if (!debugCanvas) { return; }
+            debugCanvas.width = tr.width;
+            debugCanvas.height = tr.height;
+        };
+        resize();
+
+        window.addEventListener("resize", resize);
+    }
+    return debugCanvas;
+}
+
+function drawLine(ctx: CanvasRenderingContext2D, start: Vec2, end: Vec2) {
+    ctx.beginPath();
+    ctx.moveTo(...start);
+    ctx.lineTo(...end);
+    ctx.stroke();
+}
+
+function drawPolygon(ctx: CanvasRenderingContext2D, corners: Array<Vec2>) {
+    ctx.beginPath();
+    ctx.moveTo(...corners[corners.length - 1]);
+    for (const corner of corners) {
+        ctx.lineTo(...corner);
+    }
+    ctx.stroke();
+}
+
+function drawBox(ctx: CanvasRenderingContext2D, corners: Array<Vec3>) {
+    assert(corners.length === 8, `AABB needs 8 corners, found ${corners.length}`);
+    drawPolygon(ctx, corners.slice(0, 4));
+    drawPolygon(ctx, corners.slice(4));
+    drawLine(ctx, corners[0], corners[4]);
+    drawLine(ctx, corners[1], corners[5]);
+    drawLine(ctx, corners[2], corners[6]);
+    drawLine(ctx, corners[3], corners[7]);
+}
+
+export function drawAabbs(painter: Painter, sourceCache: SourceCache, coords: Array<OverscaledTileID>) {
+    const tr = painter.transform;
+
+    const worldToECEFMatrix = mat4.invert(new Float64Array(16), tr.globeMatrix);
+    const ecefToPixelMatrix = mat4.multiply(mat4.identity(new Float64Array(16)), tr.pixelMatrix, tr.globeMatrix);
+    const numTiles =  1 << tr.coveringZoomLevel({tileSize: tr.tileSize});
+
+    if (!tr.freezeTileCoverage) {
+        aabbCorners = coords.map(coord => {
+            const aabb = aabbForTileOnGlobe(tr, numTiles, coord.canonical);
```

Comment:
instead of `numTiles` we could also pass in `tr.worldSize` and then in the for-loop avoid the scaling.

Reply:
Not sure I understand, are you suggesting reducing the number of arguments passed into `aabbForTileOnGlobe`? It's called in one other place [here](https://github.com/mapbox/mapbox-gl-js/blob/main/src/geo/projection/tile_transform.js#L99) so we'd need to make the same changes there.

Reply:
I meant to say we could pass in `tr.worldSize` instead of `numTiles`.

Reply:
Are you suggesting that we use `tr.worldSize` to calculate numTiles inside of `aabbForTileOnGlobe`?